src/app/server/OnboardingCodesUtil.cpp

```diff
@@ -97,17 +97,25 @@ CHIP_ERROR GetSetupPayload(chip::SetupPayload & aSetupPayload, chip::RendezvousI
     aSetupPayload.version               = 0;
     aSetupPayload.rendezvousInformation = aRendezvousFlags;
 
-    err = ConfigurationMgr().GetSetupPinCode(aSetupPayload.setUpPINCode);
+    err = ConfigurationMgr().GetCommissionableDataProvider()->GetSetupPasscode(aSetupPayload.setUpPINCode);
     if (err != CHIP_NO_ERROR)
     {
-        ChipLogProgress(AppServer, "ConfigurationMgr().GetSetupPinCode() failed: %s", chip::ErrorStr(err));
+        ChipLogError(AppServer, "ConfigurationMgr().GetCommissionableDataProvider()->GetSetupPasscode() failed: %s",
+                     chip::ErrorStr(err));
+#if defined(CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE) && CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE
```

Comment:
Is it needed given the legacy provider already provides this default?

Reply:
Since the legacy provider *will go away*, I didn't want this test stuff to break when it does go away. You should not rely on looking at impl of innards of the platform in application code

Reply:
Yes, but it will go away when developers provide their own implementations that no longer require this test stuff (and if they do, it can be included in the implementations).